Scripts/Runtime/Entities/Lifecycle/EntitySpawnSystem.cs

```diff
@@ -370,6 +370,13 @@ public void RegisterEntityPrototypeForDefinition<TEntitySpawnDefinition>(Entity
         public void UnregisterEntityPrototypeForDefinition<TEntitySpawnDefinition>(bool shouldDestroy)
             where TEntitySpawnDefinition : unmanaged, IEntitySpawnDefinition
         {
+            //If we're tearing down and this system was destroyed before whoever was trying to unregister, we 
+            //can't do anything. It's ok, because our Dispose would have handled freeing everything up.
+            if (m_EntityPrototypes.IsDisposed)
```

Comment:
🚲 Might want to check `this.IsDisposed` instead? Unless the disposal of the other fields is causing the call to this method.

Reply:
There is no `IsDisposed` on the system and it feels a bit weird to introduce one just for this system.